### PR TITLE
py-morphio + mpi variant & neurodamus-py deps

### PR DESCRIPTION
**Pull request by alexsavulescu**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-morphio/package.py [hunk 1]
```diff
@@ -20,8 +20,13 @@ class PyMorphio(PythonPackage):
     version('2.1.2', tag='v2.1.2', submodules=True, get_full_repo=True)
     version('2.0.8', tag='v2.0.8', submodules=True, get_full_repo=True)
 
+    variant('mpi', default=False, description="Use parallel HDF5 when needed")
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-morphio/package.py:23]: Same as the other review… why?

Comment (alexsavulescu) [reply var/spack/repos/builtin/packages/py-morphio/package.py:23]: Was having issues when I brought up `py-morphio` into `py-neurodamus`, it was linking against non-mpi flavour of `hdf5` and causing conflicts. Do you see any alternative to this?

Comment (matz-e) [reply var/spack/repos/builtin/packages/py-morphio/package.py:23]: Just requiring `depends_on('hdf5')` without any `mpi` flags. Might break a whole bunch of other stuff, though

Comment (alexsavulescu) [reply var/spack/repos/builtin/packages/py-morphio/package.py:23]: Indeed, that's why I added the variant. When we load `neurodamus-*`, we normally bring in hdf5+mpi

File: var/spack/repos/builtin/packages/py-morphio/package.py [hunk 2]
```diff
@@ -20,8 +20,13 @@ class PyMorphio(PythonPackage):
...
+
     depends_on('py-setuptools', type='build')
+    depends_on('py-setuptools-scm', type='build')
 
     depends_on('cmake@3.2:', type='build')
     depends_on('py-numpy', type='run')
-    depends_on('hdf5~mpi', type=('build', 'run'))
+    depends_on('hdf5 +mpi', type=('build', 'run'), when='+mpi')
+    depends_on('hdf5 ~mpi', type=('build', 'run'), when='~mpi')
+    depends_on('mpi', when='+mpi')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-morphio/package.py]: There's no direct dependency on MPI, so this should be nuked.

**Review by matz-e — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-morphio/package.py (hunk 1 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-morphio/package.py:23]: Then at least (for now) don't directly depend on MPI so that we can undo this later.

**Review by matz-e — APPROVED** [review]
I'm grumbling but won't stand in the way.

**Comment by alexsavulescu** [comment]
> I'm grumbling but won't stand in the way.

This will also be needed for `neuron + morphio = ❤️ `. Where we have this: 
```
    variant('mpi', default=True, description="Build with MPI support")
    ....
    depends_on('hdf5~mpi', when='~mpi')
    depends_on('hdf5+mpi', when='+mpi')
    depends_on('mpi', when='+mpi')
```
https://github.com/BlueBrain/spack/blob/develop/var/spack/repos/builtin/packages/morphio/package.py

**Review by ferdonline — APPROVED** [review]

**Comment by matz-e** [comment]
Retest this please

**Comment by matz-e** [comment]
Retest this please

File: var/spack/repos/builtin/packages/py-morphio/package.py (hunk 1 above)
Comment (pramodk) [reply var/spack/repos/builtin/packages/py-morphio/package.py:23]: I think all this `~mpi` and `+mpi` should be re-evaluated.

Comment (matz-e) [reply var/spack/repos/builtin/packages/py-morphio/package.py:23]: Yes… to be done soon™. With the direct dependency on `mpi` omitted, it's faster to grep and adjust packages.

**Pull request merged**